service/worker/worker.go

```diff
@@ -49,20 +48,19 @@ type (
 		workers          []sdkworker.Worker
 		workerComponents []workercommon.WorkerComponent
 	}
-
-	initParams struct {
-		fx.In
-		Logger           log.Logger
-		SdkClientFactory sdk.ClientFactory
-		WorkerComponents []workercommon.WorkerComponent `group:"workerComponent"`
-	}
 )
 
-func NewWorkerManager(params initParams) *workerManager {
+// NewWorkerManager creates a new worker manager. The workerComponents argument must be first in order for the fx param
+// tag to work correctly.
+func NewWorkerManager(
+	workerComponents []workercommon.WorkerComponent,
```

Comment:
why move these out of a struct into params, but in other instances move params into a struct? (e.g. deletenamespace.newComponent)

Reply:
I understand the apparent inconsistency. Let me explain:

1. **Moving Dependencies out of Structs**: For the worker manager's dependencies, I transitioned them out of a struct and into params. The primary reason was to avoid relying on the \`group:"workerComponent"\` string. Field tags must be literals, which means we couldn't use the constant WorkerComponentGroupTag. A simple typo in this string could lead to unexpected outcomes.
1. **Moving Params into Structs**: Conversely, for the worker component providers, I consolidated the parameters into a struct. The goal was to allow me to use AnnotateWorkerComponentProvider function on them, which only accepts a unary function. I wanted to restrict the AnnotateWorkerComponentProvider function to be anything that provides a `WorkerComponent`, but given Go's limited generics capabilities, I was only able to do so by also restricting the parameters of the function. So, I decided that `AnnotateWorkerComponentProvider[T any](func (t T) WorkerComponent)` was better than `AnnotateWorkerComponentProvider(interface {})`, even though it restricts what the parameters of those providers can be.